lib/datadog/core/configuration/agent_settings_resolver.rb

```diff
@@ -201,11 +200,15 @@ def timeout_seconds
         # In transport_options, we try to invoke the transport_options proc and get its configuration. In case that
         # doesn't work, we include the proc directly in the agent settings result.
         def deprecated_for_removal_transport_configuration_proc
-          if settings.tracing.transport_options.is_a?(Proc) && transport_options.adapter.nil?
-            settings.tracing.transport_options
+          if transport_options_settings.is_a?(Proc) && transport_options.adapter.nil?
+            transport_options_settings
           end
         end
 
+        def transport_options_settings
+          @transport_options_settings ||= settings.tracing&.transport_options
```

Comment:
Is safe navigation supported in all our Rubies? I think we still support 2.2 which doesn't have this.

Reply:
You're correct. CI caught it and I updated to not use safe navigation.